rust/main/agents/relayer/src/msg/metadata/base.rs

```diff
@@ -187,3 +192,189 @@ impl AppContextClassifier {
         Ok(None)
     }
 }
+
+/// An ISM caching policy.
+#[derive(Copy, Clone, Debug, PartialEq, Default, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum IsmCachePolicy {
+    /// Default cache policy, includes the message in the cache key
+    /// when querying config that may be message-specific.
+    #[default]
+    MessageSpecific,
+    /// Even if an ISM's config interface is message-specific, we
+    /// ignore the message and use the same config for all messages.
+    IsmSpecific,
+}
+
+/// Configuration for ISM caching behavior.
+/// Fields are renamed to be all lowercase / without underscores to match
+/// the format expected by the settings parsing.
+#[derive(Debug, Clone, Default, Deserialize)]
+pub struct IsmCacheConfig {
+    #[serde(deserialize_with = "deserialize_module_types", rename = "moduletypes")]
+    module_types: HashSet<ModuleType>,
+    chains: Option<HashSet<String>>,
```

Comment:
is there a noticeable reduction in RPC calls on chain where this was enabled and had a good hit rate?